### PR TITLE
adds the GOES-Test orbital slot

### PR DESCRIPTION
**Pull request by pjhartzell**
**Related Issue(s):**
#17 

**Description:**
Adds a GOES-Test orbital slot

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/stactools/goes_glm/constants.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ class Platforms(str, enum.Enum):
 class OrbitalSlot(str, enum.Enum):
     GOES_West = "West"
     GOES_East = "East"
+    GOES_Test = "Test"
```
Comment (m-mohr) [inline src/stactools/goes_glm/constants.py:89]: Test is not a valid value in the goes extension. So this is meant to be a valid value? "Test" just feels weird as a valid value.

**Comment by m-mohr** [comment]
In general this also looks like a valid approach. Having "Test" in the metadata feels a bit weird, but should be fine. 

I updated the test a bit to specifically test for "Goes-Test" instead of just cahcking the existence of the property. Also, the generated metadata files are a bit different it seems so I also added a Goes-Test based example to the examples folder.

Comparing the two examples, there are some inconsistencies in the two examples (G17 vs G17 Test) and I'd like to check that before merging. (Doing that now.)

**Comment by m-mohr** [comment]
Some notable differences:
- Test doesn't contain the variable yaw_flip_flag
- Some variables have the unit km² instead of m²
- For the geoparquet file I'm missing the time... columns - this is the most suspicious thing right now and needs a check!

**Comment by m-mohr** [comment]
I checked it and it was because millisecond offsets were not present in the other data. All offsets in Goes-Test are milliseconds while for the other datasets it is seconds. Thus, I had to fix the code to also handle that correctly. See https://github.com/stactools-packages/goes-glm/pull/21/commits/8525581e9fb1ebaa2c62025c4f8f5146e589992a for details

**Review by m-mohr — APPROVED** [review]

**Comment by pjhartzell** [comment]
Interesting. The change from milliseconds to seconds is noted in the [validation document](https://www.noaasis.noaa.gov/pdf/ps-pvr/goes-16/GLM/Full/GOES16_GLM_FullValidation_ProductPerformanceGuide.pdf) that we were looking at for other reasons. On the top of page 15:
>As an aside, the times in the prior versions of the output (prior to DO.07.00.00) have been
changed from milliseconds to seconds. Although this change is actually documented, it is still a
major change (and might require changes to the reading routines).

This suggests that non-Test data that uses milliseconds may exist. Regardless, thanks for taking care of this.

**Comment by m-mohr** [comment]
The millisecond detection is based on the actual content and not based on orbital slots in the metadata so I think this should also work in case there is non-Test data in milliseconds.

**Pull request merged**